src/npm-utils.js

```diff
@@ -121,33 +119,24 @@ module.exports = {
   },
 
   /**
-   * Return tarball URL for a given package and version.
-   * Basically, execute `npm view pkg@version dist.tarball` and return the results.
-   *
-   * As with the `releases` method, note that NPM will return an object, such as:
-   *
-   * ```json
-   *   { '1.7.7': { 'dist.tarball': 'http://registry.npmjs.org/bower/-/bower-1.7.7.tgz' } }
-   * ```
+   * Download tarball using `npm pack pkg@version`, move to temporary folder and return
+   * the location of the tarball in the temporary folder.
    *
    * The promise will be resolved with the tarball
-   * URL (i.e `'http://registry.npmjs.org/bower/-/bower-1.7.7.tgz'`).
+   * path (i.e. `'/home/user/.cache/bower-1.7.7.tgz'`).
    *
    * @param {String} pkg The package name.
    * @param {String} version The package version.
+   * @param {String} [dir] Tarball download location.
    * @return {Promise} The promise object.
    */
-  tarball: function(pkg, version) {
-    return execViewCommand([pkg + '@' + version, 'dist.tarball'])
-      .then(function(data) {
-        // If object contains the tarball URL, return it.
-        if (data['dist.tarball']) {
-          return data['dist.tarball'];
-        }
-
-        // Otherwise, unwrap it.
-        var mostRecentVersion = getLastKey(data);
-        return data[mostRecentVersion]['dist.tarball'];
+  downloadTarball: function(pkg, version, dir) {
+    var oldCWD = process.cwd();
+    process.chdir(dir);
+    return execPackCommand([pkg + '@' + version])
+      .then(function(filename) {
+        process.chdir(oldCWD);
```

Comment:
Can you move this line in the `finally` callback of the promise (to reset the process working directory even if something failed with `execPackCommand`)?